### PR TITLE
Collapse automation/script editor sections by default

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Make all triggers, conditions and actions collapsible and collapse by default.

When adding a new trigger/condition/action, it will automatically expand the newly added one.

Previews:
- [Trigger](https://deploy-preview-13390--home-assistant-gallery.netlify.app/#automation/editor-trigger)
- [Condition](https://deploy-preview-13390--home-assistant-gallery.netlify.app/#automation/editor-condition)
- [Action](https://deploy-preview-13390--home-assistant-gallery.netlify.app/#automation/editor-action)
- [Expansion panel gallery](https://deploy-preview-13390--home-assistant-gallery.netlify.app/#components/ha-expansion-panel)

This relies heavily on the `describeTrigger`, `describeCondition` and `describeAction` functions which will need to be expanded.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 1]
```diff
@@ -441,32 +461,54 @@ export default class HaAutomationActionRow extends LitElement {
...
+        }
+
         .disabled {
           opacity: 0.5;
           pointer-events: none;
         }
         .card-content {
           padding-top: 16px;
           margin-top: 0;
+          display: none;
```
Comment (balloob) [inline src/panels/config/automation/action/ha-automation-action-row.ts:495]: I used display: none instead of removing the DOM elements so the state of the DOM elements remains stored. Although I think there was a decorator for this too.

Comment (balloob) [reply src/panels/config/automation/action/ha-automation-action-row.ts:495]: https://lit.dev/docs/templates/directives/#caching-and-change-detection

**Comment by steverep** [comment]
I'll review the technicals, but but this is a welcome addition from an accessibility standpoint. It makes navigation easier for assistive technology and less distraction for people with cognitive disabilities.

**Review by steverep — CHANGES_REQUESTED** [review]
So the bad news is that the accessibility is pretty bad.  Keyboard and assistive technology users are going to get confused quick.  Basically, you want to implement the [accordion pattern in the ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/), but you're missing all the ARIA and necessary keyboard handlers.

The very good news is @bramkragten already put in some excellent a11y work creating the [`<ha-expansion-panel>`](https://github.com/home-assistant/frontend/blob/6f19ea1d8480a5ea6446bd09719fb6de26091ce3/src/components/ha-expansion-panel.ts) component, which implements the ARIA pattern nicely and has a nice big target button.  For a11y and just good UX consistency, can you switch it up to use that component?

**Comment by balloob** [comment]
Pushed up using `ha-expansion-panel`. I did had to end up adding buttons inside the summary div that expands the panel, I hope that is ok.

**Comment by balloob** [comment]
Alright, all should be good.

@steverep could you test it again?

File: src/components/ha-expansion-panel.ts [hunk 2]
```diff
@@ -132,9 +149,6 @@ class HaExpansionPanel extends LitElement {
 
       .summary-icon {
         transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1);
-        margin-left: auto;
-        margin-inline-start: auto;
-        margin-inline-end: initial;
```
Comment (balloob) [inline src/components/ha-expansion-panel.ts:137]: @bramkragten do you remember why you did this?

Comment (balloob) [reply src/components/ha-expansion-panel.ts:137]: I mean the `margin-inline-start` and `margin-inline-end`

Comment (zsarnett) [reply src/components/ha-expansion-panel.ts:137]: That is for Left to Right and Right to Left text

Comment (zsarnett) [reply src/components/ha-expansion-panel.ts:137]: @yosilevy Does a lot of these

Comment (balloob) [reply src/components/ha-expansion-panel.ts:137]: Ok, well I've added a design page now for expansion panel so it's easier to maintain.

Comment (balloob) [reply src/components/ha-expansion-panel.ts:137]: [Expansion panel gallery](https://deploy-preview-13390--home-assistant-gallery.netlify.app/#components/ha-expansion-panel)

**Comment by steverep** [comment]
@balloob I'm trying to review again and can't get the app to load in Firefox or Chrome

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 3]
```diff
@@ -172,10 +143,15 @@ export default class HaAutomationActionRow extends LitElement {
...
-        <div class="card-menu">
+        <ha-expansion-panel leftChevron>
+          <div slot="header" class="name">
+            ${describeAction(this.hass, this.action)}
+          </div>
+
           ${this.index !== 0
             ? html`
                 <ha-icon-button
+                  slot="header"
```
Comment (steverep) [inline src/panels/config/automation/action/ha-automation-action-row.ts:154]: Can't do this.  The entire header is already a button, so now you have a button inside a button which causes lots of problems.

I like the idea of being able to move them up and down without expanding (which is what I assume this does? I can't operate it.).  I think you need to create another optional slot in `ha-expansion-panel` to pass this in.

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 4]
```diff
@@ -187,6 +163,7 @@ export default class HaAutomationActionRow extends LitElement {
           ${this.index !== this.totalActions - 1
             ? html`
                 <ha-icon-button
+                  slot="header"
```
Comment (steverep) [inline src/panels/config/automation/action/ha-automation-action-row.ts:166]: Same here, and obviously the same for the triggers and conditions.

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 5]
```diff
@@ -195,7 +172,13 @@ export default class HaAutomationActionRow extends LitElement {
                 ></ha-icon-button>
               `
             : ""}
-          <ha-button-menu corner="BOTTOM_START" @action=${this._handleAction}>
+          <ha-button-menu
+            slot="header"
```
Comment (steverep) [inline src/panels/config/automation/action/ha-automation-action-row.ts:176]: Same goes here, but IMHO this should be inside the collapsed panel.  You don't need it unless you're working on it.

File: src/panels/config/automation/action/ha-automation-action-row.ts [hunk 6]
```diff
@@ -195,7 +172,13 @@ export default class HaAutomationActionRow extends LitElement {
...
+            fixed
+            corner="BOTTOM_START"
+            @action=${this._handleAction}
+            @click=${preventDefault}
```
Comment (steverep) [inline src/panels/config/automation/action/ha-automation-action-row.ts:180]: Why is this needed?  Because of the nested buttons?

**Review by steverep — DISMISSED** [review]
Aside from the line comments, I thought the goal was to be able to collapse the main regions for the triggers, conditions, and actions as well.  If not, should they be?

The changes you made to those main region headings have resulted in the help link and button not being labeled anymore.  I can investigate tomorrow if you can't figure that out.

Also what happened to the short descriptions?

Are you frustrated you made me a member yet? 😛 🤯

File: src/panels/config/automation/action/ha-automation-action-row.ts (hunk 6 above)
Comment (balloob) [reply src/panels/config/automation/action/ha-automation-action-row.ts:180]: Yes.

File: src/panels/config/automation/action/ha-automation-action-row.ts (hunk 4 above)
Comment (balloob) [reply src/panels/config/automation/action/ha-automation-action-row.ts]: Fixed

**Review by zsarnett — DISMISSED** [review]

File: src/panels/config/automation/action/ha-automation-action-row.ts (hunk 5 above)
Comment (balloob) [reply src/panels/config/automation/action/ha-automation-action-row.ts]: Excellent point. Maybe we can show it only if expanded or move it into the menu itself? But let's explore that in a future PR.

**Review by zsarnett — APPROVED** [review]

**Comment by balloob** [comment]
@steverep I addressed your latest comments. Merging now and I will follow-up to address any other concerns.
